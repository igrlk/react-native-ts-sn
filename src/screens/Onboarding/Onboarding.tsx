import React, { useEffect } from 'react';
import { View, Text } from 'react-native';

import Screen from 'library/common/commonComponents/Screen';

import styles from './onboardingStyles';
import Button from 'library/common/commonComponents/Buttons/Button';
import { getNavigateTo } from 'library/utilities/navigator';

export default function Onboarding() {
	return (
		<Screen>
			<View style={styles.container}>
				<Text style={styles.title}>1. Select Service</Text>
				<Text style={styles.text}>
					Sign up on our app, select number of samples to be diagnosed, add mailing address and pay.
				</Text>

				<Text style={styles.title}>2. Ship Us Samples</Text>
				<Text style={styles.text}>
					Complete information on sample labels. Ship us samples in our pre-paid envelope.
				</Text>

				<Text style={styles.title}>3. Get Results</Text>
				<Text style={styles.text}>You can review the diagnosis report in less than a week. </Text>

				<Button style={styles.button} onClick={getNavigateTo('Landing')}>
					Get Started
				</Button>
			</View>
		</Screen>
	);
}
