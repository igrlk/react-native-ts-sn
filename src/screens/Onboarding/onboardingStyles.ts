import EStyleSheet from 'react-native-extended-stylesheet';

export default EStyleSheet.create({
	title: {
		fontFamily: 'Poppins',
		fontSize: 24,
		lineHeight: 36,
		color: '$colorBlack',
		marginBottom: 9,
	},
	text: {
		fontFamily: 'Poppins',
		color: '$colorBlack',
		fontSize: 18,
		marginBottom: 26,
	},
	button: {
		marginTop: 20,
	},
	container: {
		marginTop: 20,
	},
});
