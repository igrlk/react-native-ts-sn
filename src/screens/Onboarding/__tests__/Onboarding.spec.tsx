import React from 'react';
import renderer from 'react-test-renderer';

import Onboarding from '../Onboarding';

describe('Onboarding', () => {
	it('match to snapshot', () => {
		const tree = renderer.create(<Onboarding />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
