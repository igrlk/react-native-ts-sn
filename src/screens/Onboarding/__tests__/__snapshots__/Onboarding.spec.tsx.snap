// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Onboarding match to snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        undefined,
        undefined,
        Object {},
      ]
    }
  >
    <View>
      <RCTScrollView>
        <View>
          <View>
            <Text>
              1. Select Service
            </Text>
            <Text>
              Sign up on our app, select number of samples to be diagnosed, add mailing address and pay.
            </Text>
            <Text>
              2. Ship Us Samples
            </Text>
            <Text>
              Complete information on sample labels. Ship us samples in our pre-paid envelope.
            </Text>
            <Text>
              3. Get Results
            </Text>
            <Text>
              You can review the diagnosis report in less than a week. 
            </Text>
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Array [
                    undefined,
                    undefined,
                  ]
                }
              >
                Get Started
              </Text>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</View>
`;
